classes/class-ep-debug-bar-elasticpress.php

```diff
@@ -162,12 +166,84 @@ public function render() {
 						<?php endif; ?>
 						<a class="copy-curl" data-request="<?php echo esc_attr( addcslashes( $curl_request, '"' ) ); ?>">Copy cURL Request</a>
 					</li><?php
-					
+
 				endforeach;
 
 			?></ol><?php
 
 		endif;
 	}
 
+	/**
+	 * Display debug info that effects the Elastic Press logs being visible or not.
+	 */
+	protected function display_debugging_info() {
+		$is_debug_errors = false;
+		?>
+		<div class="qm-boxed">
+			<section>
+				<h3><?php esc_html_e( 'Debug Info', 'debug-bar' ); ?></h3>
+				<p>
+					<?php esc_html_e( 'Elastic Press query logs not being displayed can be due to one of the following reasons:', 'debug-bar' ); ?>
+				</p>
+				<table>
+					<tbody>
+						<?php
+						$is_wp_debug = defined( 'WP_DEBUG' ) && WP_DEBUG;
+						$is_wp_ep_debug = defined( 'WP_EP_DEBUG' ) && WP_EP_DEBUG;
+
+						// Display error message only if both WP_DEBUG and WP_EP_DEBUG are turned off.
+						if ( ! ( $is_wp_debug || $is_wp_ep_debug ) ) :
+							$is_debug_errors = true;
+							?>
+						<tr>
+							<td><?php echo '&#10060;'; // cross mark. ?></td>
+							<td>
+								<?php
+								echo wp_kses(
+									sprintf( __( 'Either %s or %s should be defined and set to true', 'debug-bar' ), '<code>WP_DEBUG</code>', '<code>WP_EP_DEBUG</code>' ),
+									[ 'code' => [] ]
+								);
+								?>
+							</td>
+						</tr>
+						<?php
+						endif;
+						?>
+
+						<?php
+						// Data is being currently indexed.
+						if ( ep_is_indexing() || ep_is_indexing_wpcli() ) :
```

Comment:
Think you removed the `else` statement from this bit?

Reply:
Do we even need the `else` statement? If error is there, let's display and do nothing otherwise.

Reply:
The text is saying "Index is currently in sync" when this condition is checking to see if it's currently being indexed which isn't right - hence asking about the `else` statement being missing.

I think it's an important distinction to note that indexing going on is not an error, it's a status.

Reply:
Ah, I might have been lost in translation with `Index is currently in sync` I assumed it mean that data is being currently synchronised. I'll change the English phrase.
Do I only display the note when indexing and nothing otherwise? Or do we need a message in both cases?

Reply:
aha! Sorry - yes so if things are in sync that means they're synchronised and there's nothing to do.

I think a message in both cases is helpful to relay the current status of the index.

Reply:
Sorry - on this one you're right, just update the text and leave the logic as is, we only need to show if indexing is currently happening.